AlphaWallet/KeyManagement/EtherKeystore.swift

```diff
@@ -0,0 +1,647 @@
+// Copyright SIX DAY LLC. All rights reserved.
+
+import Foundation
+import LocalAuthentication
+import BigInt
+import KeychainSwift
+import Result
+import TrustWalletCore
+
+enum EtherKeystoreError: LocalizedError {
+    case protectionDisabled
+}
+
+///We use ECDSA keys, achieving symmetric encryption based on Diffie-Hellman to encrypt the HD wallet seed and raw private keys and store the ciphertext in the keychain
+///
+///The ECDSA key is stored in the secure enclave, not requiring user presence
```

Comment:
Why is the user presence requirement added to the ciphertext and not the ECDSA key? We are weakening security by allowing usage of the ecdsa key without authentication rather than the ciphertext. If possible, require user presence for both; if only one is possible, key is preferable to ciphertext.

Reply:
Both the ECDSA key and ciphertext now requires user presence.